lib/gui/modules/analytics.js

```diff
@@ -164,23 +87,39 @@ analytics.service('AnalyticsService', function($log, $window, $mixpanel, Setting
    * @public
    *
    * @description
-   * This function logs an exception in TrackJS.
+   * This function logs an exception to error reporting services.
    *
    * @param {Error} exception - exception
    *
    * @example
    * AnalyticsService.logException(new Error('Something happened'));
    */
   this.logException = (exception) => {
-    if (_.every([
-      SettingsModel.get('errorReporting'),
-      isRunningInAsar(),
-      errors.shouldReport(exception)
-    ])) {
-      $window.trackJs.track(exception);
-    }
+    resinCorvus.logException(exception);
+  };
 
-    $log.error(exception);
+  /**
+   * @summary Enable analytics
+   * @function
+   * @public
+   *
+   * @example
+   * AnalyticsService.enable()
+   */
+  this.enable = () => {
```

Comment:
You're right. Using `Store.subscribe(callback)` doesn't solve the issue, because by the time the callback is called, we already log events.

Besides checking the setting in `analytics` before sending events, like we did before, I don't have any good ideas.